pfioh/swift_store.py

```diff
@@ -11,195 +11,146 @@
 from   pfioh                       import base64_process, zip_process, zipdir
 from   keystoneauth1.identity      import v3
 from   keystoneauth1               import session
-from   swiftclient                 import client as swift_client
+from   swiftclient                 import service as swift_service
 from   pfmisc._colors              import Colors
+from   shutil                      import copyfileobj
+
+def _createSwiftService(configPath):
+    config = configparser.ConfigParser()
+    try:
+        f = open(configPath, 'r')
+        config.readfp(f)
+    finally:
+        f.close()
+
+    options = {
+        'auth_version':         3,
+        'os_auth_url':          config['AUTHORIZATION']['osAuthUrl'],
+        'os_username':          config['AUTHORIZATION']['username'],
+        'os_password':          config['AUTHORIZATION']['password'],
+        'os_project_domain_name':    config['PROJECT']['osProjectDomain'],
+        'os_project_name':      config['PROJECT']['osProjectName']
+    }
+
+    service = swift_service.SwiftService(options)
+    return service
+
+def _getObject(str_containerName, b_delete):
+    """
+    Returns an object associated with the specified key in the specified container
+    Deletes the object after returning if specified
+    """
+
+    str_containerName = str_key
+    str_key = os.path.join('output','data')
+    swiftDataObject = swiftConnection.get_object(str_containerName, str_key)
+    if b_delete:
+        swiftConnection.delete_object(str_containerName, str_key)
+        print('Deleted object with key %s' %str_key)
+    return swiftDataObject
+
+def storeData(**kwargs):
+    """
+    Creates an object of the file and stores it into the container as key-value object 
+    """
+
+    configPath = "/etc/swift-credentials.cfg"
+    
+    for k,v in kwargs.items():
+        if k == 'input_stream': inputStream         = v
+        if k == 'path':         str_containerName   = v
+        if k == 'is_zip':       b_zip               = v
+        if k == 'd_ret':        d_ret               = v
+        if k == 'client_path':  str_clientPath      = v
+        if k == 'configPath':   configPath          = v
+
+    swiftService = _createSwiftService(configPath)
+
+    if not b_zip:
+        with zipfile.ZipFile('/local/outgoingData.zip', 'w', compression=zipfile.ZIP_DEFLATED) as zipfileObj:
+            with zipfileObj.open(str_clientPath.split('/')[-1], 'w') as entry:
+                shutil.copyfileobj(inputStream, entry)
+    else:
+        with f as open('/local/outgoingData.zip', w):
+            shutil.copyfileobj(inputStream, f)
+
+    try:
+        success = True
+        uploadObject = swift_service.SwiftUploadObject('/local/outgoingData.zip', "input/data")
+        uploadResultsGenerator = swiftService.upload(containerName, [uploadObject])
+        for res in uploadResultsGenerator:
+            print("Upload results generated")
+            if not res["success"]:
+                success = False
```

Comment:
Fails if an exception is thrown or if the download failed

Reply:
We can simplify this by initializing `success = False` and inside `for` loop assigning `success = res["success"]` . It will save you the effort to reassignment in `except` block.  It's not required that you do this, just a suggestion.